### PR TITLE
[Reef][NFS-Ganesha][Upgrade]Add suite

### PR DESCRIPTION
**Pull request by MohitBis**
# Description

Adding reef upgrade suite for NFS-Ganesha

### CODE REVIEW

**Review by pranavprakash20 — APPROVED** [review]
LGTM

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+              nfs_version: 4.1
+              clients: 4
+        - test:
+            name: Nfs Verify File Operations
+            module: nfs_verify_file_operations.py
+            desc: Change the ownership (user and group) of files within the NFS export
+            polarion-id: CEPH-83577601
+            abort-on-fail: false
+            config:
+              nfs_version: 4.2
```
Comment (msainiRedhat) [inline suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml:186]: Version should be 4.1

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                client02 : change_ownership
+                client03 : perform_lookups
+        - test:
+            name: Nfs Verify Readdir Ops
+            module: nfs_verify_readdir_ops.py
+            desc: Perform readir operation from clients
+            polarion-id: CEPH-83577590
+            abort-on-fail: false
+            config:
+              nfs_version: 4.2
```
Comment (msainiRedhat) [inline suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml:200]: Same. Version should be 4.1

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-rhcs-7x-to-7x-latest.yaml [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+              nfs_version: 4.1
+              clients: 4
+        - test:
+            name: Nfs Verify File Operations
+            module: nfs_verify_file_operations.py
+            desc: Change the ownership (user and group) of files within the NFS export
+            polarion-id: CEPH-83577601
+            abort-on-fail: false
+            config:
+              nfs_version: 4.2
```
Comment (msainiRedhat) [inline suites/reef/nfs/tier1-nfs-ganesha-upgrade-rhcs-7x-to-7x-latest.yaml:186]: Same. Version should be 4.1

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-rhcs-7x-to-7x-latest.yaml [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                client02 : change_ownership
+                client03 : perform_lookups
+        - test:
+            name: Nfs Verify Readdir Ops
+            module: nfs_verify_readdir_ops.py
+            desc: Perform readir operation from clients
+            polarion-id: CEPH-83577590
+            abort-on-fail: false
+            config:
+              nfs_version: 4.2
```
Comment (msainiRedhat) [inline suites/reef/nfs/tier1-nfs-ganesha-upgrade-rhcs-7x-to-7x-latest.yaml:200]: Same. Version should be 4.1

**Review by msainiRedhat — CHANGES_REQUESTED** [review]

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+      parallel:
+        - test:
+            name: Nfs Verify multiple parallel io and lookups
+            module: nfs_verify_multiple_parallel_io_and_lookups.py
+            desc: Perform look ups while multiple parallel io are in progress
+            polarion-id: CEPH-83581304
+            abort-on-fail: false
+            config:
+              nfs_version: 4.1
+              clients: 4
```
Comment (msainiRedhat) [inline suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml:178]: Each Test is creating NFS cluster and deleting the cluster post performing the operations. Better to have all operations on 1 single NFS cluster without teardown and later upgrade the same NFS cluster

**Review by msainiRedhat — CHANGES_REQUESTED** [review]

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml (hunk 1 above)
Comment (MohitBis) [reply suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml]: Thanks @msainiRedhat 
Updated

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml (hunk 2 above)
Comment (MohitBis) [reply suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml]: Thanks @msainiRedhat 
Updated

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-rhcs-7x-to-7x-latest.yaml (hunk 3 above)
Comment (MohitBis) [reply suites/reef/nfs/tier1-nfs-ganesha-upgrade-rhcs-7x-to-7x-latest.yaml:186]: Thanks @msainiRedhat 
Updated

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-rhcs-7x-to-7x-latest.yaml (hunk 4 above)
Comment (MohitBis) [reply suites/reef/nfs/tier1-nfs-ganesha-upgrade-rhcs-7x-to-7x-latest.yaml:200]: Thanks @msainiRedhat 
Updated

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml (hunk 5 above)
Comment (MohitBis) [reply suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml:178]: Hi @msainiRedhat 

As of now we not have any test for just creating cluster without teardown 

As per the discussion with "Pranav" as of now we are putting existing test case, Later once we received the 7.1 build we will create test specific to upgrade functionlity

**Review by pranavprakash20 — APPROVED** [review]

File: suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml (hunk 5 above)
Comment (msainiRedhat) [reply suites/reef/nfs/tier1-nfs-ganesha-upgrade-ibm-7x-to-7x-latest.yaml:178]: Noted

**Review by msainiRedhat — APPROVED** [review]

**Review by vamahaja — APPROVED** [review]

**Pull request merged**